File: python-service/app/models/workout_plan.py
from sqlalchemy import Column, Integer, String, DateTime, ForeignKey
from app.db.database import Base
import datetime

class User(Base):
    __tablename__ = "users"
    __table_args__ = {"schema": "fitness_tracker"}
    id = Column(Integer, primary_key=True, index=True)
    username = Column(String, unique=True)
    email = Column(String, unique=True)

class WorkoutPlan(Base):
    __tablename__ = "workout_plans"
    __table_args__ = {"schema": "fitness_tracker"}

    id = Column(Integer, primary_key=True, index=True)
    user_id = Column(Integer, ForeignKey("fitness_tracker.users.id"))
    goal = Column(String)
    level = Column(String)
    days_per_week = Column(Integer)

File: java-service/src/main/java/com/fitness/fitnessapp/entity/WorkoutPlan.java
package com.fitness.fitnessapp.entity;

import jakarta.persistence.*;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.ToString;

import java.time.OffsetDateTime;
import java.util.ArrayList;
import java.util.List;

@Entity
@Table(name="workout_plans", schema="fitness_tracker")
@Data
public class WorkoutPlan {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private long id;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "user_id", nullable = false)
    @ToString.Exclude
    @EqualsAndHashCode.Exclude
    private User user;

    @Column(nullable = false)
    private String goal;

    @Column(nullable = false)
    private String level;

    @Column(nullable = false)
    private int daysPerWeek;

    @Column
    private String splitType;

    @Column(nullable = false)
    private String notes;
    @Column(nullable = false, updatable = false)
    private OffsetDateTime createdAt = OffsetDateTime.now();

    @OneToMany(mappedBy = "workoutPlan", cascade = CascadeType.ALL, orphanRemoval = true)
    @ToString.Exclude
    @EqualsAndHashCode.Exclude
    private List<WorkoutPlanItem> items = new ArrayList<>();
}
